src/webapp/pages/current-call/CurrentCallPage.tsx

```diff
@@ -1,47 +1,77 @@
+/* eslint-disable no-console */
 import { Breadcrumbs, Button, Typography } from "@material-ui/core";
 import { CircularProgress } from "material-ui";
-import React from "react";
+import React, { useState } from "react";
 import styled from "styled-components";
 import { MainLayout } from "../../components/main-layout/MainLayout";
 import { useAppContext } from "../../contexts/app-context";
 import { useGlassModule } from "../../hooks/useGlassModule";
 import { glassColors, palette } from "../app/themes/dhis2.theme";
 import ChevronRightIcon from "@material-ui/icons/ChevronRight";
-import { NavLink } from "react-router-dom";
+import { NavLink, useLocation } from "react-router-dom";
 import { CustomCard } from "../../components/custom-card/CustomCard";
 import i18n from "@eyeseetea/d2-ui-components/locales";
 import { CurrentCallContent } from "../../components/current-call/CurrentCallContent";
+import { useSpecificCall } from "../../hooks/useSpecificCall";
+import { statusMap } from "../../components/current-call/StatusMap";
 
-interface CurrentCallPageContentProps {
+interface CurrentCallPageProps {
     moduleName: string;
 }
 
-export const CurrentCallPage: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleName }) => {
-    return (
-        <MainLayout>
-            <CurrentCallPageContent moduleName={moduleName} />
-        </MainLayout>
-    );
-});
+interface CurrentCallPageContentProps {
+    moduleId: string;
+    moduleName: string;
+}
 
-export const CurrentCallPageContent: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleName }) => {
+export const CurrentCallPage: React.FC<CurrentCallPageProps> = React.memo(({ moduleName }) => {
     const { compositionRoot } = useAppContext();
 
-    // TODO: replace useGlassModule (or parameters) with actual hook to fetch current call data
     const result = useGlassModule(compositionRoot, moduleName);
+    switch (result.kind) {
+        case "loading":
+            return <CircularProgress />;
+        case "error":
+            return <Typography variant="h6">{result.message}</Typography>;
+        case "loaded":
+            return (
+                <MainLayout>
+                    <CurrentCallPageContent moduleName={moduleName} moduleId={result.data.id} />
+                </MainLayout>
+            );
+    }
+});
+
+export const CurrentCallPageContent: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleId, moduleName }) => {
+    const location = useLocation();
+    const queryParameters = new URLSearchParams(location.search);
+
+    //TO DO : orgUnit and period will not be fetched from queryParameters, it will be fetched from context.
+    const periodVal = queryParameters?.get("period");
+    const orgUnitVal = queryParameters.get("orgUnit");
+
+    //set default values till the context changes are integrated
+    const [period, setPeriod] = useState(periodVal === null ? 2023 : parseInt(periodVal));
```

Comment:
We should avoid hardcoded magic numbers (2023), maybe use the current year using a date object?

Reply:
Removed hardcoded values.